### PR TITLE
python312Packages.captum: init at 0.7.0

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
@ofborg build captum

**Comment by drupol** [comment]
@ofborg build python312Packages.captum

File: pkgs/development/python-modules/captum/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      "tests/influence/_core/test_tracin_self_influence.py"
+      "tests/influence/_core/test_tracin_show_progress.py"
+      "tests/influence/_core/test_tracin_validation.py"
+      "tests/influence/_core/test_tracin_xor.py"
+      "tests/insights/test_contribution.py"
+      "tests/module/test_binary_concrete_stochastic_gates.py"
+      "tests/module/test_gaussian_stochastic_gates.py"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      "tests/attr/test_data_parallel.py"
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/captum/default.nix:70]: Can you add a comment explaining why this test fails please ?

File: pkgs/development/python-modules/captum/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  flask-compress,
+}:
+
+buildPythonPackage rec {
+  pname = "captum";
+  version = "0.7.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/captum/default.nix:24]:
```suggestion
```

This is not really useful, especially if it is a version way older than the ones we officially support.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
@ofborg build python312Packages.captum

File: pkgs/development/python-modules/captum/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/python-modules/captum/default.nix]: OK, I disabled that part to run Ofborg, get the log, add the comment, commit and re-request you for a review :D

Comment (drupol) [reply pkgs/development/python-modules/captum/default.nix:70]: Job Done!

**Comment by drupol** [comment]
@ofborg build python312Packages.captum

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
https://paste.glepage.com/upload/bear-pug-viper

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.captum</li>
    <li>python311Packages.captum.dist</li>
    <li>python312Packages.captum</li>
    <li>python312Packages.captum.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]
LGTM

File: pkgs/development/python-modules/captum/default.nix [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  disabledTests = [
+    # Failing tests
+    "test_softmax_classification_batch_multi_target"
+    "test_softmax_classification_batch_zero_baseline"
+  ];
+
+  meta = {
+    description = "Model interpretability and understanding for PyTorch";
+    homepage = "https://github.com/pytorch/captum";
+    license = lib.licenses.bsd3;
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/captum/default.nix:86]:
```suggestion
    homepage = "https://github.com/pytorch/captum";
    changelog = "https://github.com/pytorch/captum/releases/tag/v${version}";
    license = lib.licenses.bsd3;
```

Comment (drupol) [reply pkgs/development/python-modules/captum/default.nix:86]: Zut.... Je corrige ça asap.

**Pull request merged**